### PR TITLE
Persistent bootlogo

### PR DESCRIPTION
**Pull request by mkopec**
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

### CODE REVIEW

**Comment by mkopec** [comment]
Re-marking this as a draft, since this PR still has issues reading system info (CPU, RAM, SMBIOS) and putting it in the setup menu.
Probably a conflict caused by one of the commits adding CBMEM logging.

Other than that, it boots, displays and places the logo into BGRT from the `BOOTSPLASH` region successfully.

**Comment by mkopec** [comment]
Removed the CBMEM logging patches, they should probably be merged separately. This fixed the issues in the setup menu.

File: MdeModulePkg/Universal/Acpi/BootGraphicsResourceTableDxe/BootGraphicsResourceTableDxe.c [hunk 1]
```diff
@@ -588,15 +588,15 @@ BootGraphicsDxeEntryPoint (
...
-  Status = gBS->CreateEventEx (
-                  EVT_NOTIFY_SIGNAL,
-                  TPL_CALLBACK,
-                  BgrtReadyToBootEventNotify,
-                  NULL,
-                  &gEfiEventReadyToBootGuid,
-                  &mBootGraphicsReadyToBootEvent
-                  );
-  ASSERT_EFI_ERROR (Status);
+//  Status = gBS->CreateEventEx (
```
Comment (miczyg1) [inline MdeModulePkg/Universal/Acpi/BootGraphicsResourceTableDxe/BootGraphicsResourceTableDxe.c]: No, do not modify the generic modules. If you need to remove it, do it in custom module in UEFIpaylaodPkg for example.

File: UefiPayloadPkg/UefiPayloadPkg.dec [hunk 2]
```diff
@@ -45,6 +45,7 @@
   #
   gPlatformGOPPolicyGuid                  = { 0xec2e931b, 0x3281, 0x48a5, { 0x81, 0x07, 0xdf, 0x8a, 0x8b, 0xed, 0x3c, 0x5d } }
 
+
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkg.dec]: obsolete change

File: UefiPayloadPkg/UefiPayloadPkg.fdf [hunk 3]
```diff
@@ -185,6 +186,13 @@ INF MdeModulePkg/Bus/Scsi/ScsiDiskDxe/ScsiDiskDxe.inf
 
 INF FatPkg/EnhancedFatDxe/Fat.inf
 
+#
+# ACPI Support
+#
+
+INF MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableDxe.inf
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkg.fdf]: Do we really use this? We use hacks to add BGRT to ACPI RSDP, not the AcpiTableDxe and BootGraphicsResourceTableDxe modules.

File: UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc [hunk 4]
```diff
@@ -555,7 +557,8 @@
 
   UefiCpuPkg/CpuDxe/CpuDxe.inf
   MdeModulePkg/Universal/BdsDxe/BdsDxe.inf
-  MdeModulePkg/Logo/LogoDxe.inf
+  MdeModulePkg/Logo/Logo.inf
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc]: UefiPayloadPkg/Logo/Logo.inf ?

File: UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc [hunk 5]
```diff
@@ -596,6 +599,7 @@
   MdeModulePkg/Universal/HiiDatabaseDxe/HiiDatabaseDxe.inf
   MdeModulePkg/Universal/SetupBrowserDxe/SetupBrowserDxe.inf
   MdeModulePkg/Universal/DisplayEngineDxe/DisplayEngineDxe.inf
+  MdeModulePkg/Universal/Acpi/BootGraphicsResourceTableDxe/BootGraphicsResourceTableDxe.inf
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc]: Again, do we really use this?

**Review by miczyg1 — CHANGES_REQUESTED** [review]
Please fix the comments

File: MdeModulePkg/Universal/Acpi/BootGraphicsResourceTableDxe/BootGraphicsResourceTableDxe.c (hunk 1 above)
Comment (mkopec) [reply MdeModulePkg/Universal/Acpi/BootGraphicsResourceTableDxe/BootGraphicsResourceTableDxe.c]: Change reverted, we don't need it anymore.

File: UefiPayloadPkg/UefiPayloadPkg.dec (hunk 2 above)
Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkg.dec]: deleted

File: UefiPayloadPkg/UefiPayloadPkg.fdf (hunk 3 above)
Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkg.fdf]: We don't use it anymore. I thought I needed it at one point to build everything but as it turns out it's not needed now.

Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkg.fdf]: Deleted

File: UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc (hunk 4 above)
Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc]: Yes. Changed

File: UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc (hunk 5 above)
Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc]: Nope

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**